### ISSUE TITLE
Add alt text to sponsor logo in homepage blocks.

### ISSUE THREAD

**Issue by claudiulodro** [Automattic/newspack-blocks#678]
**Describe the bug**
[When the logo for a sponsored post is built here](https://github.com/Automattic/newspack-blocks/blob/1f4880ad57e78098b719dc308f2358d5b84a0d13/src/blocks/homepage-articles/templates/article.php#L142), it should add alt text if possible. This will improve the accessibility of that element.

**To Reproduce**
1. Go to https://wave.webaim.org/report#/https://newspack-model-home.newspackstaging.com
2. If the sponsored post is still on the homepage of the model home, observe the accessibility tester complains that there is no alt text for the sponsor image.

**Screenshots**

<img width="547" alt="Screen Shot 2021-01-22 at 3 22 04 PM" src="https://user-images.githubusercontent.com/7317227/105559347-c34f5100-5cc5-11eb-909d-9a482b30798b.png">

**Expected behavior**
Accessibility checkers should not complain about the sponsor logo.
